### PR TITLE
feat(superuser): assign readonly scopes

### PR DESCRIPTION
**Pull request by cathteng**
Assign readonly scopes to a superuser user if all of the following conditions are met:

1. Superuser is active
2. The org is not self-hosted
3. The feature flag is on for the user
4. The user does not have `superuser.write` permission

For https://github.com/getsentry/team-enterprise/issues/39

### CODE REVIEW

File: tests/sentry/auth/test_access.py [hunk 1]
```diff
@@ -538,6 +538,13 @@ def test_superuser(self):
         result = self.from_request(request)
         assert result.has_permission("test.permission")
 
+    @patch("sentry.auth.access.is_active_superuser", return_value=True)
```
Comment (cathteng) [inline tests/sentry/auth/test_access.py:541]: this mock is not being respected :(

the real `is_active_superuser` function is still being called regardless if i mock using `sentry.auth.access.is_active_superuser` or `sentry.auth.superuser.is_active_superuser`

see the breakpoints

Comment (cathteng) [reply tests/sentry/auth/test_access.py:541]: in other tests i see Richard actually logged in as superuser, so i'm wondering if something weird is going on here

Comment (cathteng) [reply tests/sentry/auth/test_access.py:541]: but this test has an `is_active_superuser` mock and appears to be fine https://github.com/getsentry/sentry/blob/f03f11710b307ce7d7dbff661b6e7dfb218ecb0b/tests/sentry/api/test_base.py#L607

File: src/sentry/auth/access.py [hunk 2]
```diff
@@ -936,6 +938,7 @@ def from_request_org_and_scopes(
     scopes: Iterable[str] | None = None,
 ) -> Access:
     is_superuser = is_active_superuser(request)
+    breakpoint()
```
Comment (ykamo001) [inline src/sentry/auth/access.py:941]: Project just leftover from debugging!

Comment (cathteng) [reply src/sentry/auth/access.py:941]: yes it's leftover in case anybody wants to see what i'm seeing :'(

File: src/sentry/auth/access.py [hunk 3]
```diff
@@ -957,10 +960,15 @@ def from_request_org_and_scopes(
             org_member=member,
         )
 
+        superuser_scopes = SUPERUSER_SCOPES
+        if not is_self_hosted() and features.has("auth:superuser-read-write", member):
+            if "superuser.write" not in auth_state.permissions:
```
Comment (ykamo001) [inline src/sentry/auth/access.py:965]: nit: could we combine the 2 ifs together? Seems like they all need to be invoked together for the single condition to happen right?

File: src/sentry/auth/access.py [hunk 4]
```diff
@@ -1020,6 +1028,7 @@ def from_request(
     request: Any, organization: Optional[Organization] = None, scopes: Iterable[str] | None = None
 ) -> Access:
     is_superuser = is_active_superuser(request)
+    breakpoint()
```
Comment (ykamo001) [inline src/sentry/auth/access.py:1031]: nit: same here!

File: src/sentry/auth/access.py [hunk 5]
```diff
@@ -1044,10 +1053,15 @@ def from_request(
             org_member=(summarize_member(member) if member is not None else None),
         ).sso_state
 
+        superuser_scopes = SUPERUSER_SCOPES
+        if not is_self_hosted() and features.has("auth:superuser-read-write", member):
+            if "superuser.write" not in sso_state.permissions:
```
Comment (ykamo001) [inline src/sentry/auth/access.py:1058]: nit: same comment

Comment (ykamo001) [reply src/sentry/auth/access.py:1058]: if we need this check in multiple places, could we make a util/helper func for this that returns the proper scope?

**Review by ykamo001 — APPROVED** [review]
as long as you remove the breakpoints, you should be good to go!

File: src/sentry/auth/access.py (hunk 5 above)
Comment (cathteng) [reply src/sentry/auth/access.py:1058]: good point!

**Review by nhsiehgit — COMMENTED** [review]
quick comment

very nit though so i wont' be offended if you skip 😬

File: tests/sentry/auth/test_access.py [hunk 6]
```diff
@@ -538,6 +540,37 @@ def test_superuser(self):
...
+    def test_superuser_in_organization_scopes(self, mock_is_active_superuser):
+        request = self.make_request(user=self.superuser, is_superuser=True)
```
[the remark this answers is not in the document]
Comment (cathteng) [reply tests/sentry/auth/test_access.py:545]: good point. i don't think it's being used

**Pull request merged**